### PR TITLE
Update sidecar injector docs

### PR DESCRIPTION
**Pull request by ayj**
fixes https://github.com/istio/istio.github.io/issues/1005

### CODE REVIEW

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 1]
```diff
@@ -7,16 +7,6 @@ aliases:
...
 
-> The following requires Istio 0.5 or greater. See
-> [https://archive.istio.io/v0.4/docs/setup/kubernetes/sidecar-injection](https://archive.istio.io/v0.4/docs/setup/kubernetes/sidecar-injection)
-> for Istio 0.4 or prior.
->
-> In previous releases, the Kubernetes initializer feature was used for automatic proxy injection. This was an Alpha feature, subject to change/removal,
-> and not enabled by default in Kubernetes. Starting in Kubernetes 1.9 it was replaced by a beta feature called
-> [mutating webhooks](https://kubernetes.io/docs/admin/admission-controllers/#mutatingadmissionwebhook-beta-in-19), which is now enabled by default in
-> Kubernetes 1.9 and beyond. Starting with Istio 0.5.0 the automatic proxy injection uses mutating webhooks, and support for injection by initializer has been
-> removed. Users who cannot upgrade to Kubernetes 1.9 should use manual injection.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: nice 👍 For further reviews, has a decision been made on a minimum pin version of Kubernetes?  I saw a different review which changed the tested version to 1.10.

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 2]
```diff
@@ -59,25 +49,21 @@ Automatic injection injects at pod creation time. The controller resource is
...
-$ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@)
+$ istioctl kube-inject -f @samples/sleep/sleep.yaml@ | kubectl apply -f -
 ```
 
-`kube-inject` can also be run without access to a running Kubernetes
-cluster. Create local copies of the injection and mesh configmap.
+Alternatively, inject using local copies of the configuration.
 
-> The `istioctl kube-inject` operation may not be repeated on the output
+> NOTE: The `istioctl kube-inject` operation may not be repeated on the output
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: I believe @geeknoid was attempting to remove the usage of `NOTE: ` from the codebase.  Martin can you confirm?

For NOTEs, can you use something like this:
```
> The `istioctl kube-inject`
```

@geeknoid do you have a preference here as to handling the notes?  Seems like an icon would be warranted in the future - had a quick look at how to implement that - seems straightforward with exception of the icon itself.

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 3]
```diff
@@ -203,8 +183,9 @@ sleep-776b7bcdcd-gmvnr   1/1       Running       0          2s
 [admissionregistration.k8s.io/v1beta1#MutatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#mutatingwebhookconfiguration-v1beta1-admissionregistration)
 configures when the webhook is invoked by Kubernetes. The default
 supplied with Istio selects pods in namespaces with label `istio-injection=enabled`.
-This can be changed by modifying the MutatingWebhookConfiguration in
-`install/kubernetes/istio-sidecar-injector-with-ca-bundle.yaml`.
+This can be changed by modifying the MutatingWebhookConfiguration with `kubectl edit mutatingwebhookconfiguration istio-sidecar-injector`.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: This refers to what?  pod selection?  Could you clarify?

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 4]
```diff
@@ -203,8 +183,9 @@ sleep-776b7bcdcd-gmvnr   1/1       Running       0          2s
 [admissionregistration.k8s.io/v1beta1#MutatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#mutatingwebhookconfiguration-v1beta1-admissionregistration)
 configures when the webhook is invoked by Kubernetes. The default
 supplied with Istio selects pods in namespaces with label `istio-injection=enabled`.
-This can be changed by modifying the MutatingWebhookConfiguration in
-`install/kubernetes/istio-sidecar-injector-with-ca-bundle.yaml`.
+This can be changed by modifying the MutatingWebhookConfiguration with `kubectl edit mutatingwebhookconfiguration istio-sidecar-injector`. 
+
+> NOTE: The sidecar injector pod(s) should be restarted after modifying the mutatingwebhookconfiguration.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: @geeknoid do you have a NOTE preference - we should try to solve holistically as people seem keen to want to have that word note present - me included :)

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 5]
```diff
@@ -246,9 +227,10 @@ struct containing the list of containers and volumes to inject into the pod.
 
 ```go
 type SidecarInjectionSpec struct {
-      InitContainers []v1.Container `yaml:"initContainers"`
-      Containers     []v1.Container `yaml:"containers"`
-      Volumes        []v1.Volume    `yaml:"volumes"`
+      InitContainers   []v1.Container `yaml:"initContainers"`
+      Containers       []v1.Container `yaml:"containers"`
+      Volumes          []v1.Volume    `yaml:"volumes"`
+      ImagePullSecrets []corev1.LocalObjectReference `yaml:"imagePullSecrets"
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: missing backtick it appears

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 6]
```diff
@@ -305,22 +287,25 @@ containers:
...
+$ kubectl delete clusterrole istio-sidecar-injector-istio-system
+$ kubectl delete clusterrolebinding istio-sidecar-injector-admin-role-binding-istio-system
 ```
 
 The above command will not remove the injected sidecars from
 Pods. A rolling update or simply deleting the pods and forcing
 the deployment to create them is required.
 
-Optionally, if may be also be desirable to clean-up other resources that were created in this task. This includes the secret holding the cert/key and CSR used to sign them, as well as any namespace that was labeled for injection.
+Optionally, if may be also be desirable to clean-up other resources that were modified in this task.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: Optionally, it may also be desirable ...

**Comment by sdake** [comment]
@ayj can add `mutatingwebhookconfiguration` to spelling file to get circleci into a passing state.

Cheers
-steve

File: content/docs/setup/kubernetes/sidecar-injection.md (hunk 2 above)
Comment (geeknoid) [reply content/docs/setup/kubernetes/sidecar-injection.md]: Yes, please just use the > syntax for notes, without the word NOTE:

You can insert one of two icons at the moment:

  > {{< warning_icon >}} This is a warning note

  > {{< idea_icon >}} This is a neato idea to try out

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 7]
```diff
@@ -304,7 +304,7 @@ The above command will not remove the injected sidecars from
 Pods. A rolling update or simply deleting the pods and forcing
 the deployment to create them is required.
 
-Optionally, if may be also be desirable to clean-up other resources that were modified in this task. 
+Optionally, if may also be desirable to clean-up other resources that were modified in this task.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: if/it?

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 8]
```diff
@@ -185,7 +185,7 @@ configures when the webhook is invoked by Kubernetes. The default
 supplied with Istio selects pods in namespaces with label `istio-injection=enabled`.
 The set of namespaces in which injection is applied can be changed by editing the MutatingWebhookConfiguration with `kubectl edit mutatingwebhookconfiguration istio-sidecar-injector`. 
 
-> NOTE: The sidecar injector pod(s) should be restarted after modifying the mutatingwebhookconfiguration.
+> {{< warning_icon >}} The sidecar injector pod(s) should be restarted after modifying the mutatingwebhookconfiguration.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: no > is needed - the `{{< warning icon >}}` should be sufficient.

File: content/docs/setup/kubernetes/sidecar-injection.md [hunk 9]
```diff
@@ -202,9 +183,13 @@ sleep-776b7bcdcd-gmvnr   1/1       Running       0          2s
...
-supplied with Istio selects pods in namespaces with label `istio-injection=enabled`.
-This can be changed by modifying the MutatingWebhookConfiguration in
-`install/kubernetes/istio-sidecar-injector-with-ca-bundle.yaml`.
+supplied with Istio selects pods in namespaces with label
+`istio-injection=enabled`.  The set of namespaces in which injection
+is applied can be changed by editing the MutatingWebhookConfiguration
+with `kubectl edit mutatingwebhookconfiguration
+istio-sidecar-injector`.
+
+> {{< warning_icon >}} The sidecar injector pod(s) should be restarted after modifying the mutatingwebhookconfiguration.
```
Comment (sdake) [inline content/docs/setup/kubernetes/sidecar-injection.md]: `>` isn't needed here for a warning icon.

Comment (ayj) [reply content/docs/setup/kubernetes/sidecar-injection.md]: https://istio.io/about/contribute/writing-a-new-topic/#adding-icons-emojis says both need to be included. And I didn't find any case where both weren't used in existing content. cc @geeknoid

Comment (sdake) [reply content/docs/setup/kubernetes/sidecar-injection.md]: My bad @ayj.

**Comment by sdake** [comment]
@geeknoid can you merge - I am not in this branch's owner file.

Cheers
-steve

**Pull request merged**